cmd/ebrelayer/relayer/ethereum.go

```diff
@@ -62,51 +59,46 @@ type EthereumSub struct {
 }
 
 // NewKeybase create a new keybase instance
-func NewKeybase(validatorMoniker, mnemonic, password string) (keys.Keybase, keys.Info, error) {
-	keybase := keys.NewInMemory()
+func NewKeybase(validatorMoniker, mnemonic, password string) (keyring.Keyring, keyring.Info, error) {
+	kr := keyring.NewInMemory()
 	hdpath := *hd.NewFundraiserParams(0, sdk.CoinType, 0)
-	info, err := keybase.CreateAccount(validatorMoniker, mnemonic, "", password, hdpath.String(), keys.Secp256k1)
+	info, err := kr.NewAccount(validatorMoniker, mnemonic, password, hdpath.String(), hd.Secp256k1)
 	if err != nil {
 		return nil, nil, err
 	}
 
-	return keybase, info, nil
+	return kr, info, nil
 }
 
 // NewEthereumSub initializes a new EthereumSub
-func NewEthereumSub(inBuf io.Reader, rpcURL string, cdc *codec.Codec, validatorMoniker, chainID, ethProvider string,
-	registryContractAddress common.Address, privateKey *ecdsa.PrivateKey, mnemonic string, db *leveldb.DB, sugaredLogger *zap.SugaredLogger) (EthereumSub, error) {
+func NewEthereumSub(inBuf io.Reader, rpcURL string, validatorMoniker, chainID, ethProvider string,
+	registryContractAddress common.Address, privateKey *ecdsa.PrivateKey, mnemonic string,
+	db *leveldb.DB, sugaredLogger *zap.SugaredLogger) (EthereumSub, error) {
 
 	tempPassword, _ := password.Generate(32, 5, 0, false, false)
-	keybase, info, err := NewKeybase(validatorMoniker, mnemonic, tempPassword)
+	kr := keyring.NewInMemory()
+	hdpath := *hd.NewFundraiserParams(0, sdk.CoinType, 0)
```

Comment:
Why is NewEthereumSub newing up a FundraiserParams object?  That seems misplaced, since this function is all about starting an ethereum subscription.

Reply:
hdpath as the name of an object returned by NewFundraiserParams seems strange. Can we give this a better name?